### ISSUE TITLE
Gut installing to wrong path

### ISSUE THREAD

**Issue by CaffeinatedCM** [Ducking-Games/godotons#3]
When attempting to Install [Gut](https://github.com/bitwes/Gut/tree/godot_4) I'm having a problem where the addon extracts to `res://Gut-godot_4` instead of `res://addons`, see screenshots:

![image](https://github.com/Ducking-Games/godotons/assets/608002/24a2e288-7c5b-47fc-91b7-8a867f35de74)

![image](https://github.com/Ducking-Games/godotons/assets/608002/bcbf85fa-1f68-4684-ab3c-e0cdee09de1d)

![image](https://github.com/Ducking-Games/godotons/assets/608002/49e5b1d8-45d9-4676-8725-8beb320592dd)

**Comment by asciifaceman** [comment]
I suspect this has to do with how I was assembling filepath URIs in my first pass

```
		var archive_name: String = "%s-%s" % [addon.Name, addon.Branch]
		var archive_path: String = "user://%s.zip" % [archive_name]
		...
		>>>> Problematic
				for archived_file in reader.get_files():
					if archived_file.contains(addon.UpstreamPath):
						if archived_file.ends_with("/"): 
							var target: String = archived_file.trim_prefix("%s-%s/" % [addon.Name, addon.Branch]) 
	        <<<< Problematic
							print_rich("[color=cyan]Creating res://%s...[/color]" % [target])
							dir.make_dir_recursive("res://%s" % [target])
						else:
							var target: String = archived_file.trim_prefix("%s-%s" % [addon.Name, addon.Branch])
							var writing: FileAccess = FileAccess.open("res://%s" % [target], FileAccess.WRITE)
							writing.store_buffer(reader.read_file(archived_file))
							writing.close()
							wrote_file_count += 1
```

I'm adding gut to my test addons and will try to address this in my rework branch for next release

**Comment by CaffeinatedCM** [comment]
Potentially related, when I attempted to import the [beehave](https://github.com/bitbrain/beehave/) script_templates I got the same issue, but importing beehave itself works as expected

Config:
```
[beehave]

name="beehave"
update=true
repo="https://github.com/bitbrain/beehave"
branch="godot-4.x"
upstream_path="addons/beehave"
project_path="addons/beehave"

[beehave templates]

name="beehave templates"
update=true
repo="https://github.com/bitbrain/beehave"
branch="godot-4.x"
upstream_path="script_templates"
project_path="scripts/templates"

```

Logs:
```
Beginning Integration Run with 2 addons
Integrating Addon: beehave
Awaiting beehave...
Successfully fetched beehave
Injecting addon [source:addons/beehave -> addons/beehave]...
Creating res://addons/beehave/...
Creating res://addons/beehave/debug/...
Creating res://addons/beehave/debug/icons/...
Creating res://addons/beehave/icons/...
Creating res://addons/beehave/metrics/...
Creating res://addons/beehave/nodes/...
Creating res://addons/beehave/nodes/composites/...
Creating res://addons/beehave/nodes/decorators/...
Creating res://addons/beehave/nodes/leaves/...
Creating res://addons/beehave/utils/...
Wrote 88 files
Cleaning up user://beehave-godot-4.x.zip
Failed to delete user://beehave-godot-4.x.zip: 1
Done with beehave
Integrating Addon: beehave templates
Awaiting beehave templates...
Successfully fetched beehave templates
Injecting addon [source:script_templates -> scripts/templates]...
Creating res://beehave-godot-4.x/script_templates/...
Creating res://beehave-godot-4.x/script_templates/BeehaveNode/...
Wrote 2 files
Cleaning up user://beehave templates-godot-4.x.zip
Failed to delete user://beehave templates-godot-4.x.zip: 1
Done with beehave templates
```

**Comment by asciifaceman** [comment]
Nothing for myself the initial bug is at least partially caused by the ProjectPath not being taken into account but rather being done based on the path within the zip file (which sometimes aligns with ProjectPath thus went unseen)

**Comment by asciifaceman** [comment]
Fixed in #2 (both issues).

![image](https://github.com/Ducking-Games/godotons/assets/1619591/8b4f7b34-6fa8-4d4e-b042-302416eda02e)

Can you install Godotons from main and try now?

**Comment by asciifaceman** [comment]
![image](https://github.com/Ducking-Games/godotons/assets/1619591/88693bb1-0c4e-405f-a095-0fe78a241c8d)

resolved beehave_templates issue
